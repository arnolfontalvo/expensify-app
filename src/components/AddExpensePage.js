/*jslint node: true */
/*jshint -W079 */
'use strict';

import React from 'react';

const AddExpensePage = () => (
  <div>
    This is from my add expense component
  </div>
);

export default AddExpensePage;