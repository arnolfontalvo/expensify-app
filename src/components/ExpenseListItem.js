/*jslint node: true */
/*jshint -W079 */
'use strict';

import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount} - {createdAt}</p>    
  </div>
);

export default ExpenseListItem;