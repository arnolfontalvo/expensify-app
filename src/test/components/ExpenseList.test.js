import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render ExpenseList with expenses correctly', ()=>{
  const wrapper = new shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();  
});

test('should render ExpenseList with no expenses correctly', ()=>{
  const wrapper = new shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();  
});