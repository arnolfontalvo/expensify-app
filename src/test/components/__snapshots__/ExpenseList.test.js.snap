// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses correctly 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={"1969-12-28T00:00:00.000Z"}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={"1970-01-05T00:00:00.000Z"}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should render ExpenseList with no expenses correctly 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
